src/yarp.c

```diff
@@ -11218,42 +11235,112 @@ parse_expression_prefix(yp_parser_t *parser, yp_binding_power_t binding_power) {
             yp_heredoc_indent_t indent = parser->lex_modes.current->as.heredoc.indent;
 
             yp_node_t *node;
-            if (quote == YP_HEREDOC_QUOTE_BACKTICK) {
-                node = (yp_node_t *) yp_interpolated_xstring_node_create(parser, &parser->current, &parser->current);
-            } else {
-                node = (yp_node_t *) yp_interpolated_string_node_create(parser, &parser->current, NULL, &parser->current);
-            }
 
             parser_lex(parser);
-            yp_node_t *part;
+
+            if (parser->current.type == YP_TOKEN_HEREDOC_END) {
+                if (quote == YP_HEREDOC_QUOTE_BACKTICK) {
+                    yp_x_string_node_t *x_str_node = yp_xstring_node_create(parser, &parser->previous, &parser->previous, &parser->current);
```

Comment:
Can you pass in a `yp_token_t` for the content token instead of updating everything below? Something like
```suggestion
                    yp_node_t *node = (yp_node_t *) yp_xstring_node_create(parser, &parser->previous, &((yp_token_t) { .type = YP_TOKEN_STRING_CONTENT, .start = parser->previous.end, .end = parser->current.start }, &parser->current);
                    node->location.end = parser->previous.end;
```